src/utils.ts

```diff
@@ -164,11 +164,14 @@ function clone<T extends object>(object: T): T {
 /**
  * This method automatically checks if the prop is function or getter and behaves accordingly.
  *
- * @param object - The object to look up the getter function in its prototype chain.
- * @param prop - The property name for which to find the getter function.
- * @returns The getter function found in the prototype chain or a fallback function.
+ * @param {T} object - The object to look up the getter function in its prototype chain.
```

Comment:
The types do not need to be specified in the JSDoc comments. The type annotations define the type.